apps/accounting/api/serializers/serializers.py

```diff
@@ -110,12 +112,90 @@ class Meta:
         model = VATSetting
         fields = ["id", "rate", "description", "vat_applicable"]
 
-    def validate(self, data):
-        rate = data.get("rate")
-        if rate is not None and (rate < 0 or rate > 100):
+    def validate_rate(self, value):
+        if value is not None and (value < 0 or value > 100):
             raise ValidationError("VAT rate must be between 0 and 100.")
+        return value
 
-        return data
+    def validate(self, attrs):
+        rate = attrs.get("rate")
+        description = attrs.get("description")
+
+        request = self.context.get("request")
+        if not request or not hasattr(request, "user"):
+            raise serializers.ValidationError("Request context is required.")
+
+        client = request.user.client
```

Comment:
You are not validating the right way, request.user.client will return an error if user has no client and the immediate if statement will never work and in this case it'll never work but rather check attributes
```suggestion
        client = hasattr(request.user, "client"):
        
```